### PR TITLE
#9901 Extend REST API for InvoiceDetailItem 

### PR DESCRIPTION
**Pull request by cristinamghita**
#9901 Extend REST API for InvoiceDetailItem

### CODE REVIEW

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/invoicecandidates/impl/CreateInvoiceCandidatesService.java [hunk 1]
```diff
@@ -266,11 +269,14 @@ private NewManualInvoiceCandidate createCandidate(@NonNull final JsonCreateInvoi
 		{
 			final List<InvoiceDetailItem> invoiceDetailItems = item.getInvoiceDetailItems()
 					.stream()
-					.map(jsonDetail -> new InvoiceDetailItem(
-							jsonDetail.getSeqNo(),
-							jsonDetail.getLabel(),
-							jsonDetail.getDescription(),
-							jsonDetail.getDate()))
+					.map(jsonDetail -> InvoiceDetailItem.builder()
```
Comment (metas-ts) [inline backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/invoicecandidates/impl/CreateInvoiceCandidatesService.java:272]: please also set the orgId here; then you can avoid loading it when saving the item

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/IInvoiceCandDAO.java [hunk 2]
```diff
@@ -396,4 +397,6 @@
 	}
 
 	void invalidateUninvoicedFreightCostCandidate(OrderId orderId);
+
+	OrgId getOrgId(@NonNull InvoiceCandidateId invoiceCandidateId);
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/IInvoiceCandDAO.java:401]: not needed anymore, after you set the OrgId at the beginning

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/externallyreferenced/ExternallyReferencedCandidateRepository.java [hunk 3]
```diff
@@ -316,13 +317,29 @@ private void storeInvoiceDetailItems(final InvoiceCandidateId invoiceCandidateId
 
 	private I_C_Invoice_Detail createI_C_InvoiceDetail(final InvoiceCandidateId invoiceCandidateId, final InvoiceDetailItem invoiceDetailItem)
 	{
+		
 		final I_C_Invoice_Detail invoiceDetailEntity = newInstance(I_C_Invoice_Detail.class);
 
 		invoiceDetailEntity.setC_Invoice_Candidate_ID(invoiceCandidateId.getRepoId());
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/externallyreferenced/ExternallyReferencedCandidateRepository.java:323]: please also set invoiceDetailEntity.AD_Org_ID here

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/externallyreferenced/ExternallyReferencedCandidateRepository.java (hunk 3 above)
Comment (metas-ts) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/externallyreferenced/ExternallyReferencedCandidateRepository.java:323]: you :+1: but you didn't do it :-P

Comment (cristinamghita) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/externallyreferenced/ExternallyReferencedCandidateRepository.java:323]: already there

**Pull request merged**